test/framework/machinedeployment_helpers.go

```diff
@@ -289,3 +290,59 @@ func machineDeploymentOptions(deployment clusterv1.MachineDeployment) []client.L
 		client.MatchingLabels(deployment.Spec.Selector.MatchLabels),
 	}
 }
+
+// ScaleAndWaitMachineDeploymentInput is the input for ScaleAndWaitMachineDeployment.
+type ScaleAndWaitMachineDeploymentInput struct {
+	ClusterProxy              ClusterProxy
+	Cluster                   *clusterv1.Cluster
+	MachineDeployment         *clusterv1.MachineDeployment
+	Replicas                  int32
+	WaitForMachineDeployments []interface{}
+}
+
+// ScaleAndWaitMachineDeployment scales MachineDeployment and waits until all machines have node ref and equal to Replicas.
+func ScaleAndWaitMachineDeployment(ctx context.Context, input ScaleAndWaitMachineDeploymentInput) {
+	Expect(ctx).NotTo(BeNil(), "ctx is required for ScaleAndWaitMachineDeployment")
+	Expect(input.ClusterProxy).ToNot(BeNil(), "Invalid argument. input.ClusterProxy can't be nil when calling ScaleAndWaitMachineDeployment")
+	Expect(input.Cluster).ToNot(BeNil(), "Invalid argument. input.Cluster can't be nil when calling ScaleAndWaitMachineDeployment")
+
+	fmt.Fprintf(GinkgoWriter, "Scaling machine deployment %s/%s from %v to %v replicas\n", input.MachineDeployment.Namespace, input.MachineDeployment.Name, input.MachineDeployment.Spec.Replicas, input.Replicas)
```

Comment:
Using Ginkgo's `By` might be cleaner here.

Reply:
In all new helpers, we log like this. If you think `By` should be used in all helpers, we should track it in a separate issue.